### DISCUSSION TITLE
Question about third party library naming guidelines

### DISCUSSION THREAD

**Discussion by nagyesta** [testcontainers/testcontainers-java#4811] [Q&A] [answered]
Hi!

I am thinking about providing a java library with my work-in-progress Azure Key Vault test double container to help developers with using it. I think I could do it mostly by using `GenericContainer` and a tiny bit of extra code. Based on [this page](https://github.com/testcontainers/testcontainers-java/blob/master/docs/contributing.md#contributing-new-modules) my "own third party library" would be a good fit.

Do you have any policies/guidelines for using the name "testcontainers" in the artifact name or just the jar name etc.?

Thanks in advance!

P.S.: I am thinking about something similar to what Maven has `maven-<yourplugin>-plugin` vs. `<yourplugin>-maven-plugin` as mentioned [here](https://maven.apache.org/guides/plugin/guide-java-plugin-development.html).

**Update:** Created [a PR](https://github.com/nagyesta/lowkey-vault/pull/15/files) proposing the module to be named as `lowkey-vault-testcontainers`.

**Comment by kiview** [comment, marked as answer]
Hi @nagyesta, thanks a lot for contributing to the Testcontainers ecosystem! 🙌 
For our own artifacts, since they are part of the Testcontainers build, we don't add an additional `testcontainers` suffix, but in think in your case, this is totally fine for now. 

The class name of the container extending `GenericContainer` should be something like `${technology}Container`, so your name of `LowkeyVaultContainer` is well aligned. 

Something we like to enforce for our own modules (and we would suggest you to consider) is to have the API only use standard types. In your case, returning `TokenCredential` would violate this guideline. The better maintainable approach might be, to just expose `username` and `password` as `Strings` and let users assemble the `TokenCredential`. 

In the future, we hope we come up with a good solution to make 3rd party libs more discoverable as well, so let's stay in touch 🙂

**Reply by nagyesta** [reply]
Hi @kiview ,
thank you for your response! Also, thank you for spending time on the code review, much appreciated! I have made the small changes you have mentioned and will release soon.

Looking forward to the discoverability related updates you mentioned, it would be a huge exposure for such a small project as mine!
